lib/src/signed_video_h26x_sign.c

```diff
@@ -304,13 +303,13 @@ generate_sei_nalu(signed_video_t *self, uint8_t **payload, uint8_t **payload_sig
         tlv_list_encode_or_get_size(self, optional_tags, num_optional_tags, payload_ptr);
     payload_ptr += optional_written_size;
     size_t mandatory_written_size = 0;
-    if (mandatory_document_size > 0) {
+    if (mandatory_tags_size > 0) {
       mandatory_written_size =
           tlv_list_encode_or_get_size(self, mandatory_tags, num_mandatory_tags, payload_ptr);
+      payload_ptr += mandatory_written_size;
     }
     size_t written_size = optional_written_size + mandatory_written_size;
     SVI_THROW_IF(written_size == 0, SVI_MEMORY);
```

Comment:
This will not catch the error if we run out of memory when writing the mandatory tags.